services/src/northbound/src/main/java/org/openkilda/northbound/controller/LinkController.java

```diff
@@ -120,4 +121,21 @@ public BatchResults delLinkProps(
             @RequestBody List<LinkPropsDto> keysAndProps) {
         return linkService.delLinkProps(keysAndProps);
     }
+
+    /**
+     * Get all flows for a particular link.
+     *
+     * @return list of flows for a particular link.
+     */
+    @ApiOperation(value = "Get all flows for a particular link, based on arguments.", response = LinkPropsDto.class,
+            responseContainer = "List")
+    @GetMapping(path = "/link/flows",
+            produces = MediaType.APPLICATION_JSON_UTF8_VALUE)
+    @ResponseStatus(HttpStatus.OK)
+    public List<FlowPayload> getFlowsForLink(@RequestParam(value = "src_switch", required = false) SwitchId srcSwitch,
```

Comment:
Why we make src_switch, src_port, dst_switch and dst_port an optional arguments (even more why they are "GET" arguments)?